packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -660,25 +668,22 @@ contract ControllerHelper is UniswapControllerHelper, EulerControllerHelper, IER
                     );
                 } else if (data[i].rebalanceVaultNftType == ControllerHelperDataType.RebalanceVaultNftType.MintNewLp) {
                     // this will execute in the use case of fully closing old LP position, and creating new one
-                    ControllerHelperDataType.LpWPowerPerpPool memory mintNewLpParams = abi.decode(
+                    ControllerHelperDataType.MintAndLpParams memory mintAndLpParams = abi.decode(
                         data[i].data,
-                        (ControllerHelperDataType.LpWPowerPerpPool)
+                        (ControllerHelperDataType.MintAndLpParams)
                     );
 
-                    uint256 tokenId = ControllerHelperUtil.lpWPowerPerpPool(
+                    uint256 tokenId;
+                    (vaultId, tokenId) = ControllerHelperUtil.mintAndLp(
```

Comment:
I don't think this works because here you may have a balance of oSQTH and ETH from removing liquidity from uniswap.  Think the mintAndLp will mint the total needed to LP assuming 0 internal balance of oSQTH.  I suggested some ideas around more global variables that can be set by certain actions, could be a solution for this..lmk if makes sense.